### ISSUE TITLE
gnt-job watch following gnt-job cancel --kill can take multiple minutes

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#1030]
Originally reported of Google Code with ID 974.

```
If you run

   gnt-debug delay --submit 20
   gnt-job cancel --kill --yes-do-it [JOBID]
   gnt-job watch [JOBID]

very quickly after each other then sometimes it can happen that the `watch` takes multiple minutes to finish.

Running another `gnt-job cancel --kill --yes-do-it [JOBID]` kills the job for good with no further delays.

We believe it's due to a race in the --kill implementation.
```
Originally added on 2014-10-14 14:26:18 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
commit 2fb6a3989089618e13fe4a0efe86c5a64e9da625
Author: Klaus Aehlig <aehlig@google.com>
Date:   Tue Oct 14 17:03:17 2014 +0200

    Retry death detection after killing
    
    We cannot avoid the race on death detection after
    forcefully killing a job: the only guarantee the
    operating system gives us is that the process will
    die eventually. However, we can improve the chance
    of being able to successfully clean up a job by
    retrying death detection. Do this.
    
    Signed-off-by: Klaus Aehlig <aehlig@google.com>
    Reviewed-by: Niklas Hambuechen <niklash@google.com>

```
Originally added on 2014-10-15 08:50:13 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:37:03 +0000 UTC.

Changed State: Released
